src/v0/util/index.js

```diff
@@ -2052,17 +2052,24 @@ const getAuthErrCategoryFromStCode = (status) => {
   }
   return '';
 };
+
 const isValidInteger = (value) => {
   if (Number.isNaN(value) || !isDefinedAndNotNull(value)) {
     return false;
   }
-  // const valueToString = `${value}`;
+  const valueToString = `${value}`;
   // Use a regular expression to check if the string is a valid integer or a valid floating-point number
-  return typeof value === 'string' ? /^-?\d+(\.\d+)?$/.test(`${value}`) : false;
+  return /^-?\d+$/.test(valueToString);
```

Comment:
why did you remove typeof checking?

Reply:
```js
if (typeof value === 'number' ) {
    return true;
} else if (typeof value ===  'string') {
   return /^-?\d+$/.test(value);
} else {
  return false;
}
```

Reply:
we are converting the input value to string and then checking it through regex expression to avoid multiple type checks

Reply:
won't that be slow?

Reply:
It's a trade-off between multiple checks for data type or convert to string and then use regex .
What's your opinion

Reply:
it is just string or number check so it is not that complex in my opinion. I avoid converting to string as it will be almost costly

Reply:
Cool will do it in that way then

Reply:
Done